docs/proposals/admission_control_changes.md

```diff
@@ -0,0 +1,279 @@
+  someField:
+  - one
+  - two
+  - three
+```
+
+this won't work because of the naming collision and the type of `someField` is
+different for the two plugins. What we'd like to see instead is:
+
+1. deprecate `--admission-control-config-file`
+1. replace it with `--admission-plugin PLUGIN=CONFIG_FILE` (or an equivalent
+   setup in a structured configuration, once we've fully moved over)
+
+You would specify `--admission-plugin` once per plugin that you want to
+configure.
+
+
+## Ability to enable/disable individual plugins
+
+We want to hard-code a set of plugins that are enabled by default.
+Administrators need to be able to override the defaults, of course. To simply
+enable or disable a specific plugin, you would create a config file that looks
+like this:
+
+```yaml
+kind: DefaultAdmissionConfig
+version: v1
+enabled: false
+```
+
+The flow to determine if a plugin is enabled would look like this:
+
+1. If the plugin is on by default
+  1. If there is no config file for it, it remains enabled
+  1. Otherwise (have config file)
+    1. If it deserializes to `DefaultAdmissionConfig`, it uses the value of
+       `enabled` from the config file
+    1. Otherwise, it is up to the individual plugin to deserialize its config
+       and determine if it should be enabled
+1. Otherwise (off by default)
+  1. Same logic as above, except if there is no config file, it remains disabled
+
+
+## Refine the interface
+
+The current interface looks like this:
+
+```go
+type Interface interface {
+  Admit(a Attributes) (err error)
+  Handles(operation Operation) bool
+}
+```
+
+We'd like to break this into multiple phases, called at the appropriate times
+within the request lifecycle.
+
+```go
+type Interface interface {
+	// ApplyDefaults is executed after the request body is deserialized into an
+	// object for create and update requests. Plugins are free to modify the
+	// object as they see fit. If an error is returned, the request is marked as
+	// failed and no changes are persisted.
+	ApplyDefaults(a Attributes) error
+
+	// Authorize is executed after all plugins' ApplyDefaults methods have
+	// executed. This gives plugins a chance to decide if the request should be
+	// allowed to proceed after the object has had all of its defaults applied. If
+	// an error is returned, the request is marked as failed and no changes are
+	// persisted.
+	Authorize(a Attributes) error
+
+	// Validate is executed just prior to invoking Create, Update, or Delete on
+	// the appropriate `rest.Storage` handler. For Create and Update, Validate is
+	// called after the handler's own `Validate` method is invoked. This gives the
+	// plugin the opportunity to validate the object after the handler has
+	// validated it, but before it is persisted. For Delete, Validate is called
+	// immediately before the handler's `Delete` method is invoked. If an error is
+	// returned, the request is marked as failed and no changes are persisted.
+	Validate(a Attributes) error
```

Comment:
Given a structure close to what we have now, I'm still not certain that this can be cleanly factored.  I'd rather not have this than make a mess for images, access reviews, and authorization.  Seems like you could just let quota deal with a compensation based on a request UID.